File: data/multilang_comparison/program_cpp_vector/src/func15.cpp
#include "program_cpp_vector.hpp" 
Array* func15(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array177;
   if (pCounter > 0) {
      array177 = vars->data.at(--pCounter);
      array177->refC++;
      DEBUG_COPY(array177->id);
   } else {
      array177 = new Array();
      array177->size = 338;
      array177->refC = 1;
      array177->id = 177;
      array177->data = std::vector<unsigned int>(array177->size, 0);
      DEBUG_NEW(array177->id);
   }
   Array_param params0;
   params0.size = 1;
   std::vector<Array*> data_params0(1);
   params0.data = data_params0;
   params0.data.at(0) = array177;
   Array* array178 = func18(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array178->id);
   if(params0.size > 0) {
   	 params0.data.clear();
   }
   unsigned int loop51 = 0;
   unsigned int loopLimit51 = (50)/2 + 1;
   for(; loop51 < loopLimit51; loop51++) {
      for (int i = 0; i < array177->size; i++) {
         array177->data.at(i)++;
      }
   }
   for (int i = 0; i < array177->size; i++) {
       if (array177->data.at(i) == 17) {
           return array177;
       }
   }
   array178->refC--;
   if(array178->refC == 0) {
      array178->data.clear();
      delete array178;
      DEBUG_FREE(array178->id);
   }
   return array177;
}

File: data/multilang_comparison/program_julia/src/func15.jl
include("program_julia_head.jl")
function func15(vars::JArray_param, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array177 = nothing
   if pCounter > 1
       array177 = vars.data[pCounter]
       pCounter -= 1
       array177.refC += 1
   else
       array177 = JArray(zeros(Int, 338), 338, 1, 177)
   end
   params0 = JArray_param{Int}()
   params0.size = 1
   push!(params0.data, array177)
   params0.size += 1
   array178 = func18(params0, get_path(), loopsFactor);
   loopLimit51 = (50)/2 + 1
   for i in 1:loopLimit51 
      for i in 0:(array177.size-1) 
         array177.data[i+1] += 1
      end
   end
   for i in 1:array177.size
       if array177.data[i] == 17
               return array177
      end
   end
   array178.refC -= 1
   if array178.refC == 0
       empty!(array178.data)
       array178 = nothing
   end
   return array177;
end

